pep8.py

```diff
@@ -268,6 +270,9 @@ def blank_lines(logical_line, blank_lines, indent_level, line_number,
                 yield 0, "E301 expected 1 blank line, found 0"
         elif blank_before != 2:
             yield 0, "E302 expected 2 blank lines, found %d" % blank_before
+    elif (logical_line and not indent_level and blank_before != 2 and
+          previous_logical_toplevel.startswith(('def', 'class'))):
+        yield 0, "E305 expected 2 blank lines before, found %d" % blank_before
```

Comment:
Hmm? I'm pretty sure the condition is right: it's looking for a non-indented line preceded by more or less than two blank lines where the preceding non-blank non-indented line was a function or class definition.

It doesn't actually expect two lines after a method-level function or class, since those are only supposed to have one separating blank line according to PEP 8.